source/lac/petsc_matrix_base.cc

```diff
@@ -639,20 +639,33 @@ namespace PETScWrappers
   MatrixBase::write_ascii(const PetscViewerFormat format)
   {
     assert_is_compressed();
+    MPI_Comm comm = PetscObjectComm((PetscObject)matrix);
```

Comment:
```suggestion
    MPI_Comm comm = PetscObjectComm(reinterpret_cast<PetscObject>(matrix));
```
we use `reinterpret_cast` elsewhere - lets do it here as well for consistency.